app/models/alchemy/element.rb

```diff
@@ -321,9 +321,11 @@ def copy_nested_elements_to(target_element)
     #   Pass an element name to get previous of this kind.
     #
     def previous_or_next(dir, name = nil)
-      elements = page.elements.published.where("#{self.class.table_name}.position #{dir} #{position}")
+      dir = dir == '>' ? '>' : '<'
+      order = dir == '>' ? :asc : :desc
+      elements = page.elements.published.where("#{self.class.table_name}.position #{dir} ?", position)
```

Comment:
Sorry I may be misunderstanding something here - please feel free to correct me!

Did you mean we can pass the `dir` variable using [array conditions](http://guides.rubyonrails.org/v3.2.8/active_record_querying.html#array-conditions)? We're only using `order` later in the query on line 328.

For example:

```ruby
dir = '<'
position = 3
page.elements.published.where("#{self.class.table_name}.position ? ?", dir, position).explain
```

returns this:

```
ActiveRecord::StatementInvalid: SQLite3::SQLException: near "'<'": syntax error: SELECT "alchemy_elements".* FROM "alchemy_elements" WHERE "alchemy_elements"."page_id" = ? AND "alchemy_elements"."parent_element_id" IS NULL AND ("alchemy_elements"."position" IS NOT NULL) AND "alchemy_elements"."public" = ? AND (alchemy_elements.position '<' 3)  ORDER BY "alchemy_elements"."position" ASC
from /Users/jesse/src/alchemy_cms/vendor/bundle/ruby/2.4.0/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:91:in `initialize'
```